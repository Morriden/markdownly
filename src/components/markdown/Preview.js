import React from 'react';
import marked from 'marked';
import styles from 'github-markdown-css/github-markdown.css';
import { useMarkdown } from '../../hooks/appContext';

const Preview = () => {
  const markdown = useMarkdown();
  const __html = marked(markdown);
  return <div className={styles['markdown-body']} dangerouslySetInnerHTML={{ __html }}></div>;
};

export default Preview;
