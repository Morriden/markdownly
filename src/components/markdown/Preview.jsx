import React from 'react';
import styles from 'github-markdown-css/github-markdown.css';
import { useSelector } from '../../hooks/appContext';
import { getMarkdownPreview } from '../../selectors/markdownSelectors';
import { useParams } from 'react-router-dom';

const Preview = () => {

  const { id } = useParams() || 1234;
  const __html = useSelector(getMarkdownPreview, +id);
  
  return <div className={styles['markdown-body']} dangerouslySetInnerHTML={{ __html }}></div>;
};

export default Preview;
