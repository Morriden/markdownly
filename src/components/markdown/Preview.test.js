import React from 'react';
import { shallow } from 'enzyme';
import Preview from './Preview';
import AppProvider from '../../provider/AppProvider';

describe('Preview component', () => {
  it('renders Preview', () => {
    const wrapper = shallow(<AppProvider><Preview /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
