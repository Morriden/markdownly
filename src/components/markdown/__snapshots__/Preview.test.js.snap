// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview component renders Preview 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "currentMarkdownID": 1234,
        "markdownList": Array [
          Object {
            "body": "Look at that markdown file.",
            "id": 1234,
            "title": "whatever",
          },
          Object {
            "body": "Look at that markdown file.",
            "id": 5678,
            "title": "whatever",
          },
        ],
      },
    }
  }
>
  <Preview />
</ContextProvider>
`;
